extensions/notebook/src/book/bookTrustManager.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { BookTreeItem } from './bookTreeItem';
+import { BookModel } from './bookModel';
+
+export interface IBookTrustManager {
+	isNotebookTrustedByDefault(notebookUri: string): boolean;
+	setBookAsTrusted(bookRootPath: string): boolean;
+}
+
+export interface IBookTrustManagerWorkspaceDetails {
+	rootPath: string;
+	getConfiguration(section?: string | undefined): vscode.WorkspaceConfiguration;
+}
+
+export class BookTrustManager implements IBookTrustManager {
+
+	private static notebookConfiguration: string = 'notebook';
+	private static notebookTrustedBooksConfiguration: string = 'trustedBooks';
+	private trustedLocalBooks: Record<string, boolean> = {};
```

Comment:
I think using a dictionary makes more sense here - having the entries be properties is kind of weird